src/main/java/net/earthcomputer/multiconnect/impl/APIImpl.java

```diff
@@ -92,18 +90,18 @@ public void removeServerboundStringCustomPayloadListener(ICustomPayloadListener<
         CustomPayloadHandler.removeServerboundStringCustomPayloadListener(listener);
     }
 
-    // TODO: rewrite
-//    @Override
-//    public <T> boolean doesServerKnow(Registry<T> registry, T value) {
-//        return registry.getKey(value).map(key -> doesServerKnow(registry, key)).orElse(false);
-//    }
-//
-//    @SuppressWarnings("unchecked")
+
+    @Override
+    public <T> boolean doesServerKnow(Registry<T> registry, T value) {
+        return registry.getKey(value).map(key -> doesServerKnow(registry, key)).orElse(false);
```

Comment:
The problem with this is that the `doesServerKnow` overload in this method calls the parent method if the method below is commented out, which in turn calls this method again, leading to infinite recursion, so these methods need to be commented out together.